File: app/src/main/java/com/babestudios/companieshouse/ui/persons/PersonsActivityView.java
package com.babestudios.companieshouse.ui.persons;

import com.babestudios.companieshouse.data.model.persons.Persons;

import net.grandcentrix.thirtyinch.TiView;
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread;

interface PersonsActivityView extends TiView {

	@CallOnMainThread
	void showProgress();

	@CallOnMainThread
	void hideProgress();

	@CallOnMainThread
	void showPersons(final Persons persons);

	@CallOnMainThread
	void showNoPersons();

	@CallOnMainThread
	String getCompanyNumber();

}

File: app/src/main/java/com/babestudios/companyinfouk/ui/persons/PersonsActivityView.kt
package com.babestudios.companyinfouk.ui.persons

import com.babestudios.companyinfouk.data.model.persons.Persons

import net.grandcentrix.thirtyinch.TiView
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread

interface PersonsActivityView : TiView {

	@get:CallOnMainThread
	val companyNumber: String

	@CallOnMainThread
	fun showProgress()

	@CallOnMainThread
	fun hideProgress()

	@CallOnMainThread
	fun showError()

	@CallOnMainThread
	fun showPersons(persons: Persons)

	@CallOnMainThread
	fun showNoPersons()

}